File: Find_if_Path_Exists_in_Graph/Find_if_Path_Exists_in_Graph.go
package main

import (
	"fmt"
)

func main() {
	// n := 3
	// edges := [][]int{{0, 1}, {1, 2}, {2, 0}}
	// destination := 2
	// source := 0
	edges := [][]int{{0, 1}, {1, 2}, {2, 0}}
	source := 0
	destination := 2
	n := 3
	fmt.Println(validPath(n, edges, source, destination))
}

func validPath(n int, edges [][]int, source int, destination int) bool {
	res := make(map[int][]int)
	for _, edge := range edges {
		u := edge[0]
		v := edge[1]
		// fmt.Println(u, v)
		res[edge[0]] = append(res[edge[0]], v)
		res[edge[1]] = append(res[edge[1]], u)
	}
	fmt.Println(res)

	visited := make(map[int]bool)
	queue := make([]int, 0)
	queue = append(queue, source)
	visited[source] = true
	for len(queue) > 0 {
		val := queue[0]
		queue = queue[1:]
		temp := res[val]
		fmt.Println(temp, "..")
		for i := 0; i < len(temp); i++ {
			vct := temp[i]
			if visited[vct] == false {
				visited[vct] = true
				queue = append(queue, vct)
			}
		}
		if visited[destination] {
			return true
		}
	}

	return visited[destination]
}

File: Find_if_Path_Exists_in_Graph/Find_if_Path_Exists_in_Graph.py
from collections import defaultdict
def validPath(n,edges,source,destination):
        ls = defaultdict(list)
        for edge in edges:
            u = edge[0]
            v = edge[1]
            ls[u].append(v)
            ls[v].append(u)
        
        queue = []
        visited = [False] * (n+1)
        queue.append(source)
        visited[source] = True
        while queue:
            val = queue[0]
            queue.pop(0)
            temp = ls[val]
            for x in range(len(temp)):
                v = temp[x]
                if visited[v]==False:
                    queue.append(v)
                    visited[v]=True
        return visited[destination]

n=10
source = 5
destination = 9
edges = [[4,3],[1,4],[4,8],[1,7],[6,4],[4,2],[7,4],[4,0],[0,9],[5,4]]
print(validPath(n,edges,source,destination))